lightning/src/ln/channelmanager.rs

```diff
@@ -8002,8 +8002,30 @@ This indicates a bug inside LDK. Please report this error at https://github.com/
 		);
 	}
 
-	fn finalize_claims(&self, sources: Vec<HTLCSource>) {
-		self.pending_outbound_payments.finalize_claims(sources, &self.pending_events);
+	fn finalize_claims(&self, sources: Vec<(HTLCSource, Option<AttributionData>)>) {
+		// Decode attribution data to hold times.
+		let hold_times = sources
+			.into_iter()
+			.filter_map(|(source, attribution_data)| {
+				if let HTLCSource::OutboundRoute { ref session_priv, ref path, .. } = source {
+					let hold_times = attribution_data.map(|attribution_data| {
+						process_onion_success(
+							&self.secp_ctx,
+							&self.logger,
+							path,
+							session_priv,
+							attribution_data,
+						)
+					});
+
+					Some((source, hold_times))
+				} else {
+					None
+				}
+			})
+			.collect();
```

Comment:
Avoid the allocation by taking an `Iterator` in PendingOutboundPayments::finalize_claims`?

Reply:
Done